ssl/quic/quic_demux.c

```diff
@@ -462,6 +544,12 @@ int ossl_quic_demux_inject(QUIC_DEMUX *demux,
     if (buf_len > urxe->alloc_len)
         return 0;
 
+    assert(urxe->demux_state == URXE_DEMUX_STATE_FREE);
+
+    urxe = demux_reserve_urxe(demux, urxe, buf_len);
+    if (urxe == NULL)
+        return 0;
```

Comment:
Either the condition `if (buf_len > urxe->alloc_len)` does not make sense or the reserve call() is superfluous?